### ISSUE TITLE
Different behaviour for creation of objects and update of objects for cluster checks runners deployment

### ISSUE THREAD

**Issue by arapulido** [DataDog/datadog-operator#232]
OK, that was kind of a cryptic title for a bug report, but I couldn't find a better one, so let me explain with examples what happened:

This is using operator `0.5.0-RC2`. 

If I deploy this configuration for the first time in a 1 node cluster:

```
apiVersion: datadoghq.com/v1alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  credentials:
    apiSecret:
      secretName: datadog-secret
      keyName: api-key
    appSecret:
      secretName: datadog-secret
      keyName: app-key
  agent:
    image:
      name: "datadog/agent:latest"
  clusterAgent:
    image:
      name: "datadog/cluster-agent:latest"
    config:
      clusterChecksEnabled: true
```

I correctly get 1 pod for the node agent, one pod for the cluster agent and 1 pod for the cluster checks runner:

```
NAME      ACTIVE   AGENT             CLUSTER-AGENT     CLUSTER-CHECKS-RUNNER   AGE
datadog   True     Running (1/1/1)   Running (1/1/1)           Running (1/1/1)         112s
```

But, if I first apply the same configuration, but without enabling cluster checks, like this one:

```
apiVersion: datadoghq.com/v1alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  credentials:
    apiSecret:
      secretName: datadog-secret
      keyName: api-key
    appSecret:
      secretName: datadog-secret
      keyName: app-key
  agent:
    image:
      name: "datadog/agent:latest"
  clusterAgent:
    image:
      name: "datadog/cluster-agent:latest"
```

so, as expected, only get 1 node agent and 1 cluster agent:

```
NAME      ACTIVE   AGENT             CLUSTER-AGENT     CLUSTER-CHECKS-RUNNER   AGE
datadog   True     Running (1/1/1)   Running (1/1/1)                           2m57s
```

But, if I now edit the object and add the cluster checks option back, so it looks like the original object:

```
apiVersion: datadoghq.com/v1alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  credentials:
    apiSecret:
      secretName: datadog-secret
      keyName: api-key
    appSecret:
      secretName: datadog-secret
      keyName: app-key
  agent:
    image:
      name: "datadog/agent:latest"
  clusterAgent:
    image:
      name: "datadog/cluster-agent:latest"
    config:
      clusterChecksEnabled: true
```

The operator knows that there has been an update and start updating the existing objects, but it won't create a cluster checks runner replica:

```
NAME      ACTIVE   AGENT              CLUSTER-AGENT      CLUSTER-CHECKS-RUNNER   AGE
datadog   True     Updating (1/1/0)   Updating (2/1/1)                                                          9m30s
```

```
NAME                                      READY   STATUS    RESTARTS   AGE
datadog-agent-pzhh2                       1/1     Running   0          29s
datadog-cluster-agent-59958b4dff-whwqr    1/1     Running   0          34s
```

**Describe what you expected:**

I would expect the reconciliation loop to work in a way that sees that that configuration requires 1 cluster check runner and will create it when updating the object

**Comment by arapulido** [comment]
It looks like with the first configuration, the clusterChecksRunner deployment is not created by the clusterChecksRunner controller. If after having the first clusterChecksRunner running then, I change the description to:

```
apiVersion: datadoghq.com/v1alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  credentials:
    apiSecret:
      secretName: datadog-secret
      keyName: api-key
    appSecret:
      secretName: datadog-secret
      keyName: app-key
  agent:
    image:
      name: "datadog/agent:latest"
  clusterAgent:
    image:
      name: "datadog/cluster-agent:latest"
    config:
      clusterChecksEnabled: true
  clusterChecksRunner:
    replicas: 2
```

I end up with 3 replicas of the clusterchecksrunner, instead of two

**Comment by arapulido** [comment]
It looks like in the first case, the `datadog-cluster-checks-runner` deployment is not created by the cluster-checks-runner controller? (as the logs I get are different from the ones I get if I put the `spec.clusterChecksRunner` section.)

**Comment by juliogreff** [comment]
Closing as #235 has been merged both to master and 0.5.
